Lib/asyncio/tasks.py

```diff
@@ -334,6 +334,11 @@ def wait_for(fut, timeout, *, loop=None):
     if timeout is None:
         return (yield from fut)
 
+    if timeout <= 0:
+        fut = ensure_future(fut, loop=loop)
+        fut.cancel()
```

Comment:
Can you verify that the new code works the same way as the current asyncio code when `fut` is an already completed Future.